java/arcs/core/entity/EntityBase.kt

```diff
@@ -427,9 +436,12 @@ private fun toReferencable(value: Any, type: FieldType): Referencable = when (ty
     throw NotImplementedError("[FieldType.Tuple]s cannot be converted to references.")
   is FieldType.ListOf ->
     (value as List<*>).map {
-      toReferencable(it!!, type.primitiveType)
+      requireNotNull(toReferencable(it, type.primitiveType)!!) {
```

Comment:
Remove the `!!` here (that's what the `requireNotNull` does).

Reply:
Whoops, thanks Cameron